ingestion/src/metadata/ingestion/source/database/deltalake/connection.py

```diff
@@ -12,117 +12,88 @@
 """
 Source connection handler
 """
+from dataclasses import dataclass
+from functools import singledispatch
 from typing import Optional
 
-import pyspark
-from delta import configure_spark_with_delta_pip
-from pyspark.sql import SparkSession
-
 from metadata.generated.schema.entity.automations.workflow import (
     Workflow as AutomationWorkflow,
 )
+from metadata.generated.schema.entity.services.connections.database.datalake.s3Config import (
+    S3Config,
+)
+from metadata.generated.schema.entity.services.connections.database.deltalake.metastoreConfig import (
+    MetastoreConfig,
+)
+from metadata.generated.schema.entity.services.connections.database.deltalake.storageConfig import (
+    StorageConfig,
+)
 from metadata.generated.schema.entity.services.connections.database.deltaLakeConnection import (
     DeltaLakeConnection,
-    MetastoreDbConnection,
 )
-from metadata.ingestion.connections.builders import get_connection_args_common
 from metadata.ingestion.connections.test_connections import test_connection_steps
 from metadata.ingestion.ometa.ometa_api import OpenMetadata
 
 
-def get_connection(connection: DeltaLakeConnection) -> SparkSession:
-    """
-    Create connection
-    """
+@dataclass
+class DeltalakeClient:
+    def __init__(self, client, config) -> None:
+        self.client = client
+        self.config = config
+
+
+@singledispatch
+def get_deltalake_client(connection, config):
```

Comment:
if this called:

```python
`get_deltalake_client(None, None`
```

then nothing will happen...

I suggest adding more exception handling.. why not raise `NotImplementedError` in the default case?

Reply:
I think it makes total sense!